### PR TITLE
Simu pp200 run14 pythia8 hfjets

### PR DESCRIPTION
**Pull request by klendathu2k**
Addition of fast jet filter for HF jet simulation.

Fix to StarGenerator/DECAY -- advance past any pythia "system" entries when checking decay chains for validity.

Setup, macros  and workflow for HF D0-jet simulation.

### CODE REVIEW

**Review by genevb — APPROVED** [review]
Honestly, pretty much all of this PR is outside of my familiarity and I'm not very confident about what I'm approving here. Trusting Jason on this one.

**Review by starsdong — APPROVED** [review]
Trust Jason on this development. This has been used for embedding.

**Comment by starsdong** [comment]
Dmitri, if you don't object, let us also merge this into main. Thanks

File: StRoot/StarGenerator/macros/production/setup.simulation.y2014x.pythia8.HFjets
Comment (plexoos) [inline StRoot/StarGenerator/macros/production/setup.simulation.y2014x.pythia8.HFjets:14]: Shouldn't this directory be created first?
This script does not seem like it can be used out of the box.

File: StRoot/StarGenerator/macros/starsim.pythia8.C
Comment (plexoos) [inline StRoot/StarGenerator/macros/starsim.pythia8.C]: Does this code compile?

File: mgr/Conscript-standard
Comment (plexoos) [inline mgr/Conscript-standard]: The config variable provided by the package should be preferred

https://github.com/star-bnl/star-sw/blob/11f02d137a851c7cb00a6ec945dd0451b1d46ebc/mgr/Conscript-standard#L770-L774

File: StRoot/StarGenerator/macros/production/simulation.y2014x.pythia8.HFjets.input.list
Comment (plexoos) [inline StRoot/StarGenerator/macros/production/simulation.y2014x.pythia8.HFjets.input.list:1]: Is it possible to run the submitted macros as is without regenerating these files?

File: StRoot/StarGenerator/macros/production/setup.simulation.y2014x.pythia8.HFjets
Comment (klendathu2k) [reply StRoot/StarGenerator/macros/production/setup.simulation.y2014x.pythia8.HFjets:14]: This macro assumes the user has checked the code out of my repository, and builds.  Once the pull request goes through, this is for documentation purposes only.

File: StRoot/StarGenerator/macros/starsim.pythia8.C
Comment (klendathu2k) [reply StRoot/StarGenerator/macros/starsim.pythia8.C]: No, it won't compile.  But surprisingly, CINT doesn't care about this typo (as long as you don't  follow the pp:W branch of the code).  Updating.

File: StRoot/StarGenerator/macros/production/simulation.y2014x.pythia8.HFjets.input.list
Comment (klendathu2k) [reply StRoot/StarGenerator/macros/production/simulation.y2014x.pythia8.HFjets.input.list:1]: Those files are an input dependency to the job.  The format of the files are:

Run#   Event#   Vx   Vy   Vz...

Updating the macro with the specification.   These files would need to be regenerated when comparing against any production of the data... (any change which alters vertex finding would require a new set of inputs).

File: StRoot/StarGenerator/macros/production/setup.simulation.y2014x.pythia8.HFjets
Comment (plexoos) [reply StRoot/StarGenerator/macros/production/setup.simulation.y2014x.pythia8.HFjets:14]: That is a pretty strong assumption if you think about the long term code preservation and reproducibility.

File: StRoot/StarGenerator/macros/production/simulation.y2014x.pythia8.HFjets.input.list
Comment (plexoos) [reply StRoot/StarGenerator/macros/production/simulation.y2014x.pythia8.HFjets.input.list:1]: If you explain the format of the input files and how to regenerate them then there is not much point in including this particular list which, as you say, need to be regenerated for other data anyway.

**Comment by starsdong** [comment]
Victor, it seems this PR needs your approval before merging. If you can take a look and comment/approve it. Thanks

**Review by perevbnlgov — APPROVED** [review]
I do not see anything suspicious

**Comment by genevb** [comment]
This PR appears to have failed the AutoBuild. I've opened issue #324 .

**Comment by klendathu2k** [comment]
This should be resolved by PR#325.  Details in the commit message.  But
the short summary is that I broke the build when I reorganized the
source tree (moved FastJetFilter one level higher).  Cons no longer
matched the directory name, and did not apply the include for the
fast jet library.

(We were already in production for the simulation at that point, so
this did not impact the results).

**Pull request merged**
